### ISSUE TITLE
VM created through ISO cannot be reinstalled through templates

### ISSUE THREAD

**Issue by div8cn** [apache/cloudstack#3603]
Through an example created by ISO, an error occurred in Reinstall VM (API: restore Virtual Machine): "Invalid ISO id provided to restore the VM"

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.11.3
4.12
4.13
~~~

##### OS / ENVIRONMENT

CentOS7 KVM+RBD primary storage

##### STEPS TO REPRODUCE

~~~
Create an instance with the name VM1 through ISO

Turn VM1 reinstall into other templates
~~~

##### EXPECTED RESULTS

~~~
VM1 is successfully rebuilt as a specified template system
~~~

##### ACTUAL RESULTS

~~~
Error returned:
Invalid ISO id provided to restore the VM
~~~

**Comment by DaanHoogland** [comment]
@div8cn I would expect reinstall to always be based on the original template. I am not sure if it should be supported for ISO based VMs. This actually sounds like a feature request to me. Can you expand a little on your scenario?

**Comment by DaanHoogland** [comment]
@div8cn can you please expand on what you expect to happen as re-install would not be different from a new install, especially in the case of ISOs.

**Comment by rhtyd** [comment]
This is expected I think.

**Comment by rohityadavcloud** [comment]
Problem statement - we can't fix the issue, maybe improve the error msg.

**Comment by shwstppr** [comment]
Fixed with #7407
